Detections/MultipleDataSources/SuspiciousVMInstanceCreationActivity.yaml

```diff
@@ -0,0 +1,138 @@
+        EntityIp,
+        Tactics,
+        Techniques,
+        ProviderName,
+        AlertTime= bin(AlertTimeGenerated, 1d),
+        AccountObjectId
+    // Join with GCPAuditLogs for VM instance creation
+    | join kind=inner (
+        GCPAuditLogs
+        | where ServiceName == "compute.googleapis.com" and MethodName endswith "instances.insert"
+        | extend
+            GCPUserUPN= tostring(parse_json(AuthenticationInfo).principalEmail),
+            GCPUserIp = tostring(parse_json(RequestMetadata).callerIp),
+            GCPUserUA= tostring(parse_json(RequestMetadata).callerSuppliedUserAgent),
+            VMStatus =  tostring(parse_json(Response).status),
+            VMOperation=tostring(parse_json(Response).operationType),
+            VMName= tostring(parse_json(Request).name),
+            VMType = tostring(split(parse_json(Request).machineType, "/")[-1])
+        | where GCPUserUPN !has "gserviceaccount.com"
+        | where VMOperation == "insert" and isnotempty(GCPUserIp) and GCPUserIp != "private"
+        | project
+            GCPOperationTime=TimeGenerated,
+            VMName,
+            VMStatus,
+            MethodName,
+            GCPUserUPN,
+            ProjectId,
+            GCPUserIp,
+            GCPUserUA,
+            VMOperation,
+            VMType
+        )
+        on $left.EntityIp == $right.GCPUserIp 
+    // Join with IdentityInfo to enrich user identity details
+    | join kind=inner (IdentityInfo 
+        | distinct AccountObjectId, AccountUPN, JobTitle
+        )
+        on AccountObjectId 
+    // Calculate the time difference between the alert and VM creation for further analysis
+    | extend TimeDiff= datetime_diff('day', AlertTime, GCPOperationTime),Name = split(GCPUserUPN, "@")[0], UPNSuffix = split(GCPUserUPN, "@")[1]
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: GCPUserIp
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: Name
+      - identifier: UPNSuffix
+        columnName: UPNSuffix
+customDetails:
+  AlertName: AlertName
+  AlertProDuctName: ProductName
+  AlertUserName: AccountUPN
+  AlertUserObjectId: AccountObjectId
+  AlertIds: SystemAlertIds
+  AlertIp: EntityIp
+  GCPUserAgent: GCPUserUA
+  GCPVMName: VMName
+  GCPProjectId: ProjectId
+  GCPVMType: VMType
+  CorrelationWith: "GCPAuditLogs"
+alertDetailsOverride:
+  alertDisplayNameFormat: "IP address {{GCPUserIp}} Assocated with {{AlertName}} found in GCP VM creation event by {{GCPUserUPN}}"
+  alertDescriptionFormat: "This detection correlates '{{ProductName}}' Alert IP addresse Entity found in VM instance creation in GCP {{ProjectId}}. It identifies successful compute instance creation, from suspicious IP addresse. By joining these datasets on network entities and IP addresses, it detects unauthorized Initial access attempts across GCP environments."
+  alertSeverityColumnName: AlertSeverity
+  alertDynamicProperties:

```

Comment:
What do you think about setting the alert severity based on the original Microsoft alert severity?

Reply:
Yes, That Could be varies based on Microsoft Alerts